sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/misc.scala

```diff
@@ -161,3 +163,197 @@ case class Crc32(child: Expression) extends UnaryExpression with ImplicitCastInp
     })
   }
 }
+
+/**
+ * A function that calculates hash value for a group of expressions.
+ */
+case class Hash(children: Seq[Expression]) extends Expression {
```

Comment:
Is it possible to remove the code duplication with BaseGenericInternalRow.hashCode?